src/lib/WalletStorage.ts

```diff
@@ -50,6 +55,25 @@ function createLocalStorageInterface(): IWalletStorage {
     remove: (keys) => new Promise(
       (resolve) => resolve(localStorage.removeItem(composeStorageKeys(keys))),
     ),
+    /**
+     * Sync state between the app and the offscreen tab. `watch` should not be
+     * called in the app. Only in the offscreen tab.
+     */
+    watch: (keys, callback) => {
+      if (!IS_OFFSCREEN_TAB) {
+        return;
+      }
+      window.addEventListener('storage', (event) => {
+        if (
+          composeStorageKeys(keys) === event.key
+          || composeStorageKeys(keys).includes(event.key!)
```

Comment:
I don't think it's reasonable to check if strings are equal or part of it is equal. The second test is enough.